### PR TITLE
Ports ambition objectives system from Skyrat

### PR DESCRIPTION
**Pull request by dzahlus**
NOT THE HUGBOXED VERSION!!!

## About The Pull Request
PORTS: https://github.com/Skyrat-SS13/Skyrat13/pull/2266
I can't believe Silicons made me code, what it says on the tin

## Why It's Good For The Game
Silicons told me to do this, that's why it's good for the game. No seriously though we're having a lot of issues with antags being too boring, lame, what have you, because of their absolute inability to accept the fact that greentext is not what it is about. Hopefully this will change everything as it allows you to set your own objectives modularily. (admins have to approve any objective changes) You still spawn with all objectives and greentext still works the same way it does, but now you can request objective removal/completion, new objectives of your choice and ambitions.

C/P from the original PR

> Adds an ambitions system for antags. It stores a series of player-set information with the purpose of describing what they hope to achieve for the round, to provide better context to their actions and gimmicks at the round end.
> 
> Admins can see and edit the antag's ambitions through the objectives-ambitions panel.
> 
> All additions, edits and deletes are logged. Players have a small 5 seconds cooldown between edits to prevent spam, admins don't.
> 
> The max number of ambitions is set to 5 by default, but it's a config and thus can be changed either during the game or on the configs.txt
> 
> Added a safe lazylist cut macro for convenience.
> 
> Adds some cooldown systems for convenience. (cit already has this)
> 
> Adds a pair of admin macros to give them the choice of teleporting to the incident reported.
> 
> Did minor cleanup on the code.
> 
> Fixes a typo on the blackmail objective. (cit doesn't have this, not included)
> 
> On the same verb/panel, added an interface for players to request modifications, removals or additions of their objectives. The cooldown for using it is 10 minutes, which is pretty steep but it was what was requested of me.
> 
> Admins can access this interface through the Player Panel, which is now in nice dark-mode, instead of blinding white.
> 
> Players get notified when the cooldown has ended, and if they have pending requests that the next one will bwoink the admins.

WIP by the way lol, please check this code and tell me if you see anything that could break shit this is as of yet untested.

EDIT: Tested locally, all functions as intended, it compiles, admin shit works, rolling objectives works etc. removing WIP tag.

## Changelog
:cl:
add: Added ambition objective system
del: Removed old objective generation
tweak: tweaked the entire objective system
refactor: refactored a bunch of shit to work with ambition
config: added configs for ambition system
admin: changes to antag panel for ambition system
/:cl:

### CODE REVIEW

**Comment by timothyteakettle** [comment]
reference the original pr and actually say what the system is

**Comment by timothyteakettle** [comment]
alright read through it and I can't see how this is anything but extremely counterproductive given responses we have seen with things such as free objectives and the removal of greentext, this just outright won't work well on this server

**Comment by dzahlus** [comment]
> alright read through it and I can't see how this is anything but extremely counterproductive given responses we have seen with things such as free objectives and the removal of greentext, this just outright won't work well on this server

doesn't remove greentext at all, it literally just gives you prayer+, the ability to request completion/removal of objectives and addition of new ones. 

see: http://prntscr.com/10dxei3

**Comment by shellspeed1** [comment]
I like this idea. Now I can set really dumb objectives like steal as many donuts as possible.

**Comment by ghost** [comment]
in general i actually want this to go through because it allows for very funny possibilities!!
but when it comes down to making antags less "boring" i don't think it will cut it. imo it covers edge cases, that's about it

**Comment by shellspeed1** [comment]
Really antags being boring comes down to the creativity of the person that is the antag. This lets people create their own objectives and maybe will give admins an idea of what they are doing in order to make shit even more funny.

File: code/__DEFINES/cooldowns.dm [hunk 1]
```diff
@@ -63,6 +65,8 @@
 
 #define COOLDOWN_START(cd_source, cd_index, cd_time) (cd_source.cd_index = world.time + cd_time)
 
+#define COOLDOWN_CHECK(cd_source, cd_index) (cd_source.cd_index < world.time)
```
Comment (LetterN) [inline code/__DEFINES/cooldowns.dm]: dupe of cd finish

**Review by LetterN — CHANGES_REQUESTED** [review]

File: code/__DEFINES/cooldowns.dm (hunk 1 above)
Comment (dzahlus) [reply code/__DEFINES/cooldowns.dm]: i sure hope this doesn't break anything

**Comment by DeltaFire15** [comment]
Merge conflicting again.

**Comment by DeltaFire15** [comment]
Marking PR as stale due to its long period without commits despite being marked WIP.

**Comment by silicons** [comment]
taking this over once rules are done in the next few days

**Comment by DeltaFire15** [comment]
Marking this as stale again due to it having no commits since may.

**Comment by DeltaFire15** [comment]
Closing this due to staleness.

**Comment by Linzolle** [comment]
Is there any word on this at all? Kevin reopened this and removed the stale but there's been no work on it since long before then.

**Comment by keronshb** [comment]
> Is there any word on this at all? Kevin reopened this and removed the stale but there's been no work on it since long before then.

It works as far as I know.  He had me do a test of it and it complies with the code.

If it's implemented, I don't know.  But as far as this exists, it doesn't have any errors or conflicts.

**Comment by Linzolle** [comment]
If you want to actually fix the conflicts on this, open it again. Closing.

**Pull request closed without merging**